js/framework/application.js

```diff
@@ -445,15 +456,25 @@ var Application = Class.inherit({
     _showViewImpl: abstract,
 
     _obtainViewLink: function(viewInfo) {
-        var key = viewInfo.key;
+        var key = viewInfo.key,
+            hash = this._viewLinksHash[key];
```

Comment:
Naming. It's not actually a hash (the key is a hash). It's rather a 'view link' (hash -> viewLink)